---
title: "Nicholas Eagles"
image: "nick.jpg"
links:
  - label: LinkedIn
    url: "https://www.linkedin.com/in/nick-eagles-7a5b30161/"
  - label: GitHub
    url: "https://github.com/Nick-Eagles"
  - label: Email
    url: "mailto:nickeagles77@gmail.com"
output:
  postcards::trestles
---
 
## Bio

I'm currently a data scientist and bioinformatician in the
[team](https://lcolladotor.github.io/bioc_team_ds/) led by Leonardo
Collado-Torres at [LIBD](https://www.libd.org/).

Here, I use R, Python, shell scripting, and more to develop computational
pipelines for processing genomic data, explore and implement machine-learning
methods, and publish research papers showcasing open-source software and biological
findings. My work in genomics spans topics such as bulk-RNA sequencing (for
which I published [SPEAQeasy](https://doi.org/10.1186/s12859-021-04142-3)),
whole-genome bisulfite sequencing (see
[BiocMAP](https://doi.org/10.1101/2022.04.20.488947)), spatial transcriptomics
(see [visiumStitched](https://doi.org/10.1186/s12864-024-10991-y)), and
cell-type deconvolution.

I also help
[install and maintain software](https://github.com/LieberInstitute/jhpce_mod_source)
for LIBD and collaborators, and mentor and share my knowledge through
[data science guidance sessions](https://lcolladotor.github.io/bioc_team_ds/data-science-guidance-sessions.html)
and [R Stats Club](http://research.libd.org/rstatsclub/) presentations. Some example
topics I've taught include
[implementing machine-learning models in python](https://youtu.be/-qFz7PwWEeY),
[system-wide software installations on high-performance computing clusters with Lmod](https://youtu.be/8w6JDsrgqVs),
and [running PyTorch-based software with GPUs at HPC environments](https://youtu.be/zOMUXYHtVJM).

## Portfolio

### Data Science Snippets

Here I showcase actual code I've developed for data-science tasks throughout
my career.

- **Building machine-learning models**: [here](https://github.com/LieberInstitute/spatialDLPFC/blob/main/code/spot_deconvo/02-cellpose/10-explore_models.py)
I use `scikit-learn` to train and test some candidate cell-type-classification
models.

- **Data wrangling, cleaning, and visualization**: Beginning with several messy,
large datasets, I use `dplyr` in R to [clean and integrate them.](https://github.com/LieberInstitute/Habenula_Pilot/blob/b7b670eafbd10093f94dafc62c787199d1fbd719/code/17_eQTL/08_beta_plots.R#L315-L374)
Along the way, I [visualize key patterns](https://github.com/LieberInstitute/Habenula_Pilot/blob/b7b670eafbd10093f94dafc62c787199d1fbd719/code/17_eQTL/08_beta_plots.R#L229-L246)
with `ggplot2`.

- **Deep learning**: In a personal project to build a neural-network-powered
chess engine, I use Keras/Tensorflow to [build and fit a CNN-based model.](https://github.com/Nick-Eagles/Chess-Engine/blob/master/external/fit_real_data.py)

### Bioinformatics Pipelines

These are *Nextflow*-based pipelines for processing genomic data where I was the
lead developer.

- **SPEAQeasy**: bulk RNA-seq preprocessing workflow, quantifying features like
genes into `SummarizedExperiment` R objects (
[code](https://github.com/LieberInstitute/SPEAQeasy) |
[documentation](http://lieberinstitute.github.io/SPEAQeasy) |
[paper](https://doi.org/10.1186/s12859-021-04142-3))

- **BiocMAP**: preprocessing workflow for bisulfite sequencing data, quantifying
methylation in `bsseq` R objects (
[code](https://github.com/LieberInstitute/BiocMAP) |
[documentation](http://research.libd.org/BiocMAP/) |
[paper](https://doi.org/10.1101/2022.04.20.488947))

## Education

**IBM** | Online

Data Science Professional Certification | October 2024 - January 2025

**University of Maryland, Baltimore County** | Baltimore, MD

B.S. in Mathematics | September 2013 - May 2018

## Experience

**Lieber Institute for Brain Development** | Research Associate II- Data Science | November 2018 - Present

## Publications

[Integrating gene expression and imaging data across Visium capture areas with visiumStitched](https://doi.org/10.1186/s12864-024-10991-y)

- Eagles et al., *BMC Genomics*, 2024.

[BiocMAP: a Bioconductor-friendly, GPU-accelerated pipeline for bisulfite-sequencing data](https://doi.org/10.1186/s12859-023-05461-3)

- Eagles et al., *BMC Bioinformatics*, 2023.

[SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor-powered RNA-seq analyses.](https://doi.org/10.1186/s12859-021-04142-3)

- Eagles et al., *BMC Bioinformatics*, 2021.

My full list of publications is available at my [ORCID profile](https://orcid.org/0000-0002-9808-5254).

## Public Presentations

**Bioconductor** | *Bioconductor* | 2023

- Presented a workshop: [A Bioconductor-style differential expression analysis powered by SPEAQeasy](http://research.libd.org/SPEAQeasyWorkshop2023/) recorded as a [video](https://www.youtube.com/watch?v=6hJXPw9PM1o)

- Presented a short talk: [Spot Deconvolution in the Post-Mortem Human DLPFC](https://speakerdeck.com/nickeagles/bioc-2023-spot-deconvolution) recorded as a [video](https://www.youtube.com/watch?v=YA5H2XP6jjU&t=700s)

**Biological Data Science** | *Cold Spring Harbor Laboratory* | 2022

- Presented a poster: [Benchmarking spot-level cell-type deconvolution methods using Visium immunofluorescence benchmark data on the human dorsolateral prefrontal cortex](https://speakerdeck.com/nickeagles/bds-2022).

**European Bioconductor** | *Bioconductor* | 2020

- Presented my [SPEAQeasy paper](https://www.youtube.com/watch?v=tPSG23hUHSI).

